hack/cluster.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -xe
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+
+OUTPUT_DIR=${OUTPUT_DIR:-${SCRIPT_DIR}/../.output}
+
+OVN_KUBERNETES_REPO="${OVN_KUBERNETES_REPO:-https://github.com/ovn-org/ovn-kubernetes}"
+OVN_KUBERNETES_BRANCH="${OVN_KUBERNETES_BRANCH:-master}"
+OVN_KUBERNETES_DIR=${OUTPUT_DIR}/ovn-kubernetes
+
+# from https://github.com/kubernetes-sigs/kind/releases
+KIND_BIN=${OUTPUT_DIR}/kind
+
+export KUBECONFIG=${OUTPUT_DIR}/kubeconfig
+
+cluster_name=virt-ipam
+op=$1
+shift
+
+function ensure_ovn_kubernetes() {
+    if [ -d "${OVN_KUBERNETES_DIR}" ]; then
+        return 0
+    fi
+    (   
+        cd ${OUTPUT_DIR}
+        git clone --depth 1 --single-branch ${OVN_KUBERNETES_REPO} -b ${OVN_KUBERNETES_BRANCH}
+    )
+}
+
+function ensure_kind() {
+    if [ -f ${KIND_BIN} ]; then
+        return 0
+    fi
+    local kind_version=v0.20.0
+    local arch=""
+    case $(uname -m) in
+        x86_64)  arch="amd64";;
+        aarch64) arch="arm64" ;;
+    esac
+    local kind_url=https://kind.sigs.k8s.io/dl/$kind_version/kind-linux-${arch}
+    curl --retry 5 -Lo ${KIND_BIN} ${kind_url}
+	chmod +x ${KIND_BIN}
+}
+
+function up() {
+    mkdir -p ${OUTPUT_DIR}
+    ensure_ovn_kubernetes
+    ensure_kind
+    kind delete cluster --name $cluster_name
+    (
+        cd ${OVN_KUBERNETES_DIR}
+        ./contrib/kind.sh --local-kind-registry -ic -ikv -i6 -mne -cn ${cluster_name}
```

Comment:
We need to find a way to ensure we don't forget about this. 

Or maybe this option about requiring all comments to be resolved in not a good thing to use ...

Let's give it a thought.

Reply:
dont worry, we must do it, i will do it for the CNAO
we also need to delete the CRD or not deploy it but this is part of OVN repo
(reason is that we want CNAO to deploy it)

Reply:
@oshoval maybe add a know to the CNAO configuration to "deploy" the crd ? 

upstream we don't have a way to install it. downstream we do. 

This is the cheapest way I can think of to be honest. Not saying it is a really good option though ...

Reply:
CNAO already knows to deploy the CRD if thats what you mean (both U/S and D/S already)

only problem is that since we use kind.sh of OVN-k8s it already install it for us
taking the liability of CNAO, and we want CNAO to do it
i can just delete it or add a flag to kind.sh to not deploy it and use that

Reply:
Downstream it will be the openshift networks operator that installs the CRD, it is not only about kind on upstream OVN-K.

Hence, the simplest way (IMO) is to have a knob to indicate if we want to install the CRD on CNAO or not (which we would use on KubeVirt upstream installation that used CNAO). Does it make sense ? 

Having it opt out would require us to patch HCO downstream to "skip" the CRD installation, thus, I think it is worse.

Reply:
np 
we just said at the past that CNAO install CRD in both cases
if you want it will do it only on U/S is is easy

yet we will need to hack about the double install when using kind.sh but this is other issue (not need to be discussed in this thread fwiw)

Reply:
btw it also means something weird about this repo
that it deploys the latest (since deploy kubevirt is enabled on kind.sh) and then cluster-sync the PR itself
which might keep leftovers theoretically

anyhow if desired we can handle it later